src/System.Numerics.Tensors/src/System/Numerics/Tensors/Experimental/IReadOnlyTensor.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface IReadOnlyTensor<T>
+    {
+        long Rank { get; }
+
+        T this[params long[] indices] { get; }
+        T this[ReadOnlySpan<long> indices] { get; }
```

Comment:
What about `params Index[] indices`?  Wasn't tensor a key target for the Index design?

Reply:
All initial consumers are still targeting full framework.

Reply:
But we're doing this in corefx?

Reply:
The current plan is for the `System.Numerics.Tensors` package to be OOB and to ship prior to netcoreapp3.0/netstandard2.1.

All current initial consumers are multi-targeting full-framework and some older netcoreapp version which doesn't support Index/Range; so we cannot support Index/Range.

Reply:
Not sure how this is language related. The issue here is support in framework. The language doesn't care where Index / Range comes from, only that it exists. If this were supported in framework then C# would just pick it up. 

Yes there are ways that we could make Index / Range better on older frameworks by lowering adoption burden of consuming APIs. The root issue though here appears to be the bigger it's just not available there.

Reply:
> Not sure how this is language related

Simply that the language design around having types for the index and range concepts was, if I remember correctly, largely motivated by a scenario like that for Tensor.  And now that scenario isn't using the thing that was designed for it.

Reply:
That's true. It was part of the initial inspiration but there were other motivating samples. Again though there is nothing we can do about tensors *decision* to not use the types after asking for the feature.